src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteBuildConfiguration.java

```diff
@@ -720,29 +725,44 @@ public void performWaitForBuild(BuildContext context, Handle handle)
         throws InterruptedException, IOException
     {
         String jobName = handle.getJobName();
-        BuildData buildData = getBuildData(handle.getQueueId(), context);
-        handle.setBuildData(buildData);
 
-        context.logger.println("  Remote build URL: " + buildData.getURL());
-        context.logger.println("  Remote build number: " + buildData.getBuildNumber());
+        RemoteBuildInfo buildInfo = handle.getBuildInfo();
+        String queueId = buildInfo.getQueueId();
+        if (queueId == null) {
+            throw new AbortException(String.format("Unexpected status: %s. The queue id was not found.", buildInfo.toString()));
+        }
+        context.logger.println("  Remote job queue number: " + buildInfo.getQueueId());
+
+        if (buildInfo.getQueueStatus() == RemoteBuildQueueStatus.QUEUED) {
```

Comment:
I'll write some utility functions like  `buildInfo.isQueued()`
`
boolean isQueued(){
  return buildInfo.getQueueStatus() == RemoteBuildQueueStatus.QUEUED;
}
`
So as the rest of comparisons of `RemoteBuildQueueStatus` & `RemoteBuildStatus`

By doing this, we can 
1. Remove the duplicate comparisons like  line 736 & 740
2. Better encapsulation